Adding a CONTRIBUTING.md file (#259)
apollographql/apollo-mcp-server: CONTRIBUTING.md

Author: I'll reference the one in router specifically. I was told to not create our own.

![ci workflow status](https://img.shields.io/github/actions/workflow/status/apollographql/apollo-mcp-server/ci.yml)
![release binaries workflow status](https://img.shields.io/github/actions/workflow/status/apollographql/apollo-mcp-server/release-bins.yml?label=release%20binaries)
![release container workflow status](https://img.shields.io/github/actions/workflow/status/apollographql/apollo-mcp-server/release-container.yml?label=release%20container)
![version](https://img.shields.io/github/v/release/apollographql/apollo-mcp-server)
![license](https://img.shields.io/github/license/apollographql/apollo-mcp-server)

## How to contribute to Apollo MCP Server

#### Bug Reporting

> * **Ensure the bug was not already reported** by searching on GitHub under [Issues](https://github.com/apollographql/apollo-mcp-server/issues) as well as the [Apollo Community forums](https://community.apollographql.com/latest).

Reviewer: We should add a big `> [!WARNING]` here to *NOT* report security issues as a bug, but instead to look at our  `SECURITY.md` to reach out to our team.

Author: Ok updated this to point to the security policy.

* If you're unable to find an open issue addressing the problem, [open a new one](https://github.com/apollographql/apollo-mcp-server/issues/new). Be sure to include a **title and clear description**, as much relevant information as possible, and a **code sample** or an **executable test case** demonstrating the expected behavior that is not occurring.

> * If appropriate add the most relevant label but leave empty if unsure.

Reviewer: ```suggestion
* If appropriate, add the most relevant label but leave empty if unsure.
```

> #### **Did you write a patch that fixes a bug?**

Reviewer 2: The items in this list seem to be a step by step guide, should the bullets be numbers instead?

Author: Was not intended to be a step by step guide, so that was just coincidental.

> * Refer to the simple branching guide for the project.

Reviewer 2: Are we missing a link here?

* Open a new GitHub pull request with the patch.
* Ensure the PR description clearly describes the problem and solution. Include the relevant issue number if applicable.

> * Before submitting, please read the [Contributing to Apollo MCP Server](#contributing-to-apollo-mcp-server) guide to know more about coding conventions, branching strategies, etc.

Reviewer 2: The text "to know more about ..." reads weird to me. Perhaps we could have something like

```suggestion
* Before submitting, please read the [Contributing to Apollo MCP Server](#contributing-to-apollo-mcp-server) guide to learn more about our coding conventions, branching strategies, etc.
```

#### **Do you intend to add a new feature or change an existing one?**

> * Suggest your change as a new [issues](https://github.com/apollographql/apollo-mcp-server/issues) using the `enhancement` label and start writing code.

Reviewer 2: ```suggestion
* Suggest your change as a new [issue](https://github.com/apollographql/apollo-mcp-server/issues) using the `enhancement` label and start writing code.
```

* You can also suggest changes and features using the [Apollo Community forums](https://community.apollographql.com/latest).

> * Before submitting, please read the [Contributing to Apollo MCP Server](#contributing-to-apollo-mcp-server) guide to know more about coding conventions, branching strategies, etc.

Reviewer 2: The items in this list don't seem to suggest any state where the potential contributor opens a PR, so this feels out of place.

Author: I can add a reference to it but this was not intended to be a step by step guide. More a collection of useful information.

#### **Do you have questions about the code or about Apollo MCP Server itself?**

* Ask any question about Apollo MCP Server using either the [issues](https://github.com/apollographql/apollo-mcp-server/issues) page or the [Apollo Community forums](https://community.apollographql.com/latest). 
* If using the issues page, please use the `question` label.

Thanks!

Apollo MCP Server team

---

### Contributing to Apollo MCP Server

#### Branching strategy

> The Apollo MCP Server project follows a more pseudo GitFlow branch strategy.

Reviewer 2: Do we need the word "more" in this sentence? Also may be worth including a link to a guide on GitFlow for those who may be unfamiliar

Author: Sure, I'll update this.

![ci workflow status](https://img.shields.io/github/actions/workflow/status/apollographql/apollo-mcp-server/ci.yml)
![release binaries workflow status](https://img.shields.io/github/actions/workflow/status/apollographql/apollo-mcp-server/release-bins.yml?label=release%20binaries)
![release container workflow status](https://img.shields.io/github/actions/workflow/status/apollographql/apollo-mcp-server/release-container.yml?label=release%20container)
![version](https://img.shields.io/github/v/release/apollographql/apollo-mcp-server)
![license](https://img.shields.io/github/license/apollographql/apollo-mcp-server)

## How to contribute to Apollo MCP Server

> #### Bug Reporting

Author: Done
